In [2]:
import pandas as pd

Задача 40: Работать с файлом california_housing_train.csv, который находится в папке sample_data. 
Определить среднюю стоимость дома, где кол-во людей от 0 до 500 (population).

In [3]:
df = pd.read_csv('california_housing_train.csv')

In [5]:
df[
    (df['population'] > 0) & (df['population'] < 500)].median_house_value.mean()

206683.83635227982

Средняя стоимость дома с жителями менее 500 = 206683.84

Задача 42: Узнать какая максимальная households в зоне минимального значения population.

In [7]:
min_population = df.population.min()
df[df['population'] == min_population].households.max()

4.0

Максимальное количество домашних хозяйств при минимальном населении = 4. 